vispy/gloo/program.py

```diff
@@ -301,7 +301,10 @@ def __setitem__(self, name, data):
             elif kind == 'attribute':
                 # Is this a constant value per vertex
                 is_constant = False
-                isscalar = lambda x: isinstance(x, (float, int))
+
+                def isscalar(x):
+                    return isinstance(x, (float, int))
+
```

Comment:
No, they aren't always bad, they are just not good when you are trying to assign a function to a variable name directly -- that is better done through `def`. See:

http://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def